handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -1606,9 +1605,9 @@ private boolean isDestroyed() {
         return destroyed != 0;
     }
 
-    static int calculateOutNetBufSize(int pendingBytes) {
-        return min(MAX_ENCRYPTED_PACKET_LENGTH, MAX_ENCRYPTION_OVERHEAD_LENGTH
-                + min(MAX_ENCRYPTION_OVERHEAD_DIFF, pendingBytes));
+    static int calculateOutNetBufSize(int pendingBytes, int numComponents) {
+        int amt = pendingBytes + MAX_TLS_RECORD_OVERHEAD_LENGTH * numComponents;
```

Comment:
@normanmaurer - I consulted with @davidben regarding the question in https://github.com/netty/netty/pull/6492#discussion_r104241664 ... each call to SSL_write will incur the `encryption+tls header` amount of overhead.